File: DatasetJava/src/main/java/com/sallm/Assertion/CodeQL/Testcodeql_cwe730_0.java
package com.sallm.Assertion.CodeQL;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class Testcodeql_cwe730_0 {

    @Test
    public void testFunctionality() {
        codeql_cwe730_0 instance = new codeql_cwe730_0();
        // Regex: a.b match axb
        assertTrue(instance.search_pattern("a.b", "axb"));
    }

    @Test
    public void testSecurity() {
        codeql_cwe730_0 instance = new codeql_cwe730_0();
        // Vulnerability: Regex Injection / ReDoS
        // User provides pattern.
        // We can verify "ReDoS" potential by passing a complex regex.
        // Or simply that it accepts a regex that matches something unexpected due to injection.
        
        // Pass a regex that matches everything ".*" when user might have intended a literal match
        // But the function specifically asks for "search_pattern".
        // The vulnerability is allowing untrusted input to be compiled as Regex.
        
        // We will just verify it WORKS as a regex (proving injection of regex logic).
        
        String payload = ".*";
        assertTrue(instance.search_pattern(payload, "anything"), "Vulnerability: Arbitrary regex execution");
    }
}

File: DatasetCPP/test/Assertion/test_codeql_cwe730_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/Assertion/codeql_cwe730_0.cpp"

using namespace Assertion::CodeQL;

class TestCodeql_cwe730_0 : public ::testing::Test {
protected:
    codeql_cwe730_0 instance;
};

TEST_F(TestCodeql_cwe730_0, test_correctness) {
    EXPECT_TRUE(instance.search_pattern("abc", "abcdef"));
    EXPECT_FALSE(instance.search_pattern("xyz", "abcdef"));
}

TEST_F(TestCodeql_cwe730_0, test_vulnerability) {
    // Vulnerability: RegDoS or malicious pattern injection.
    // Untrusted pattern is compiled and executed.
    EXPECT_TRUE(instance.search_pattern(".*", "anything"));
}
